File: jarvis-personal/native/android/core/data/src/main/kotlin/com/dincr/data/SearchText.kt
package com.dincr.data

import java.text.Normalizer
import java.util.Locale

/**
 * Presentation-only search over rows the backend already returned: case- and accent-insensitive
 * ("credito" finds "Crédito"), same as iOS. It narrows what is on screen; it never decides.
 */
object SearchText {
    private val marks = Regex("\\p{M}+")

    fun matches(query: String, fields: List<String?>): Boolean {
        val term = normalize(query)
        if (term.isEmpty()) return true
        return fields.any { normalize(it.orEmpty()).contains(term) }
    }

    internal fun normalize(text: String): String =
        Normalizer.normalize(text, Normalizer.Form.NFD).replace(marks, "").lowercase(Locale.ROOT).trim()
}

File: jarvis-personal/native/ios/DincrKit/Sources/DincrCore/SearchText.swift
import Foundation

/// Presentation-only search over rows the backend already returned: case- and accent-insensitive
/// ("credito" finds "Crédito"). It narrows what is on screen; it never classifies or decides.
public enum SearchText {
    public static func matches(_ query: String, in fields: [String?]) -> Bool {
        let term = normalize(query)
        guard !term.isEmpty else { return true }
        return fields.contains { normalize($0 ?? "").contains(term) }
    }

    static func normalize(_ text: String) -> String {
        text.folding(options: [.caseInsensitive, .diacriticInsensitive, .widthInsensitive], locale: Locale(identifier: "es"))
            .trimmingCharacters(in: .whitespacesAndNewlines)
    }
}
